src/GraphQL/Types/Scalars/DateOnlyGraphType.cs

```diff
@@ -0,0 +1,61 @@
+#if NET6_0_OR_GREATER
+
+using System;
+using System.Globalization;
+using GraphQL.Language.AST;
+
+namespace GraphQL.Types
+{
+    /// <summary>
+    /// The Date scalar graph type represents a year, month and day in accordance with the ISO-8601 standard.
+    /// Format is `yyyy-MM-dd`.
+    /// </summary>
+    public class DateOnlyGraphType : ScalarGraphType
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DateOnlyGraphType"/> class.
+        /// </summary>
+        public DateOnlyGraphType()
+        {
+            Description = "The `Date` scalar type represents a year, month and day in accordance with the " +
+                "[ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) standard. Format is `yyyy-MM-dd`";
+        }
+
+        /// <inheritdoc/>
+        public override object? ParseLiteral(IValue value) => value switch
+        {
+            NullValue _ => null,
+            StringValue stringValue => ParseDate(stringValue.Value),
+            _ => ThrowLiteralConversionError(value)
+        };
+
+        /// <inheritdoc/>
+        public override object? ParseValue(object? value) => value switch
+        {
+            DateOnly _ => value, // no boxing
```

Comment:
What about `DateTime` support here? Also what about supporting `DateOnly` for `DateGraphType`? I think that `DateGraphType` and `DateOnlyGraphType` now are some sort of twins.

Reply:
I would not parse DateTime here. If someone receives DateTime that would be a coding error. We don’t want to allow people to write errors in new code.

Reply:
Why coding error?

Reply:
This is only a matter of ease of use. It seems to me quite natural that this scalar type will support several .NET types. Semantics is not broken.

Reply:
No, the scalar type should match the variable type.  There is absolutely no reason for the DateOnly scalar type to support any other .NET type as a variable.  If someone can think of a legitimate reason, we can change it upon request.

Reply:
> No, the scalar type should match the variable type. 

What means _match_ ?
![изображение](https://user-images.githubusercontent.com/21261007/144891217-8e61d3a0-5497-41e3-b9cc-8ce7ad3866b0.png)

Reply:
`IntGraphType` accepts `ushort` variable thought the is "dedicated" `UShortGraphType`. The same is true for a plethora of other types.

Reply:
Right, because it is not known what System.Text.Json will deserialize a number into.  In this case the value stored as json is a string.  It is not going to be DateTime, DateOnly, or any other .NET type.

Reply:
Regardless, does not mean that DateTime should be coerced to DateOnly.

Reply:
i would prefer that DateGraphType only supported dateonly and DateOnlyGraphType would not need to exist. but i could not work out how to do that gracefully. if it was my project i would drop support for older fwks and only support net6

Reply:
Well, it will be completely radically 😄 . So... I give up, let it remain as it is.